torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -656,7 +664,11 @@ Tensor clamp_backward(const Tensor & grad, const Tensor &self, const Tensor& min
   // clamp: gradients not defined on min and max, so we return the subgradient 1 for these cases.
   if (max.defined() && min.defined()) {
     auto zero = at::scalar_tensor(0., grad.options());
-    return where((self >= min).logical_and_(self <= max), grad, zero);
+    if (!areAnyTensorSubclassLike({self, min, max})) {
+      // fastpath
+      return where((self >= min).logical_and_(self <= max), grad, zero);
```

Comment:
We could reduce code duplication by having intermediary result and ` ? : ;` kind of checks no?

Reply:
Yup